ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+  block:
+    - name: Make sure needed packages are installed
+      package:
+        state: present
+        name: "{{ item }}"
+      loop:
+        - "{{ packages_registry | default([]) }}"
+      become: true
+    
+    - name: open firewall port {{ registry_port }}
+      firewalld:
+        port: "{{ registry_port }}/tcp"
+        permanent: yes
+        immediate: yes
+        state: enabled
+      become: true
+    
+    - name: query getent for homedir
+      getent:
+        database: passwd
+        key: "{{ ansible_ssh_user }}"
+    
+    - name: set homedir fact
+      set_fact:
+        homedir: "{{ getent_passwd | json_query('*[4]') | first }}"
+    
+    - name: Create directory to hold the registry files
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        mode: u=rwX,g=rX,o=rX
+        recurse: yes
+      loop:
+        - "{{ registry_dir_auth }}"
+        - "{{ registry_dir_cert }}"
+        - "{{ registry_dir_data }}"
+      become: true
+    
+    - name: Generate htpasswd entry
+      command: htpasswd -bBn dummy dummy
+      register: htpass_entry
+    
+    - name: Write htpasswd file
+      copy:
+        content: '{{ htpass_entry.stdout }}'
+        dest: "{{ registry_dir_auth }}/htpasswd"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: Create registry auth for pullsecret
+      set_fact:
+        disconnected_auth: '  {"{{ ansible_fqdn }}:{{ registry_port }}": {"auth": "ZHVtbXk6ZHVtbXk=", "email": "{{ ansible_user }}@redhat.com" } }'
+  
+    - name: Write auth for disconnected to registry host
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "{{ homedir }}/{{ registry_auth_file }}"
+  
+    - name: Write auth for disconnected to localhost
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "~/{{ registry_auth_file }}"
+      delegate_to: localhost
+  
+    - name: append auth to pullsecret
+      shell: |
+        echo '{{ pullsecret }}' | jq -c \
+          '.auths += {{ disconnected_auth }}'
+      register: new_pullsecret
+  
+    - name: set pullsecret with new auth
+      set_fact:
+        pullsecret: "  {{ new_pullsecret.stdout }}"
+  
+    - name: Write pullsecret to registry host
+      copy:
+        content: '{{pullsecret}}'
+        dest: "{{ homedir }}/pullsecret.txt"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: generate ssl certs
+      shell: |
+        openssl req \
+          -newkey rsa:4096 \
+            -nodes \
+            -sha256 \
+            -keyout {{ registry_dir_cert }}/domain.key \
+            -x509 -days 365 \
+            -out {{ registry_dir_cert }}/domain.crt \
+            -subj "/C={{ cert_country }}/ST={{ cert_state }}/L={{ cert_locality }}/O={{ cert_organization }}/OU={{ cert_organizational_unit }}/CN={{ ansible_fqdn }}"
```

Comment:
can we use this instead? https://docs.ansible.com/ansible/latest/modules/openssl_certificate_module.html

Reply:
i could not get that to generate a good certificate for me. maybe i missed something.